scripts/delete-federation.sh

```diff
@@ -28,41 +28,45 @@ NS=federation-system
 
 PUBLIC_NS=kube-multicluster-public
 
-# Remove the federation service account for the current context.
-CONTEXT="$(kubectl config current-context)"
-SA_NAME="${CONTEXT}-${CONTEXT}"
-${KCD} -n ${NS} sa "${SA_NAME}"
-${KCD} -n ${NS} clusterrole "${NS}-controller-manager:${SA_NAME}"
-${KCD} -n ${NS} clusterrolebinding "${NS}-controller-manager:${SA_NAME}"
-
-CONTEXT="$(kubectl config current-context)"
-
-# Remove permissive rolebinding that allows federation controllers to run.
-${KCD} clusterrolebinding federation-admin
+# Remove public namespace
+${KCD} namespace ${PUBLIC_NS}
 
-# Remove role and role binding added by kubefed
-CONTROLLER_ROLE="federation-controller-manager:${CONTEXT}-${CONTEXT}"
-${KCD} clusterrolebinding "${CONTROLLER_ROLE}"
-${KCD} clusterrole "${CONTROLLER_ROLE}"
+# Remove cluster registry CRD
+${KCD} -f vendor/k8s.io/cluster-registry/cluster-registry-crd.yaml
 
-# Delete federated cluster
-# TODO(marun) Remove when federated clusters are stored in federation namespace
-if kubectl get federatedcluster &>/dev/null; then
-  ${KCD} federatedcluster "${CONTEXT}"
-fi
+# Unjoin clusters by removing objects added by kubefed.
```

Comment:
We can remove this once `unjoin` finished.

Reply:
I don't know that we'll necessarily want to remove this as it's part of cleanup. But the script can be simplified.